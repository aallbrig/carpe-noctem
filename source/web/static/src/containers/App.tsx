import * as React from 'react';
import { connect } from 'react-redux';
import { IInjectedProps, Link} from 'react-router';

class App extends React.Component<IInjectedProps, void> {
  render() {
    const { children } = this.props;
    return (
      <div>
        <h4>Basic Containing App.tsx File</h4>
        <div>
          { children }
        </div>
      </div>
    );
  };
};
export default connect(
  () => ({}),
  () => ({})
)(App);