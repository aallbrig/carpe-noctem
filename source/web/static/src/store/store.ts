import { createStore } from 'redux';
import { default as reducers } from '../reducers';
const reduxResponsive = require('redux-responsive');
const { calculateResponsiveState } = reduxResponsive;
import { debounce } from 'lodash';

const devTools = (window as any).__REDUX_DEVTOOLS_EXTENSION__;
export const store = createStore(
  reducers,
  devTools && devTools()
);

store.dispatch(calculateResponsiveState(window));
const onResize = debounce((): void => store.dispatch(calculateResponsiveState(window)), 1000);
window.addEventListener('resize', onResize);

export default store;
