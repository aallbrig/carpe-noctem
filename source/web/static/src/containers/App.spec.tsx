import * as React from 'react';
import { App } from './App';
import { Provider } from 'react-redux';
import { store } from '../store/store'
import * as renderer from 'react-test-renderer';

let app:renderer.ReactTestInstance;

beforeEach(() => {
  app = renderer.create(
    <Provider store={ store }>
        <App children={null} />
    </Provider>
  );
});

test('App Snapshot', () => {
  expect(app.toJSON()).toMatchSnapshot();
});
