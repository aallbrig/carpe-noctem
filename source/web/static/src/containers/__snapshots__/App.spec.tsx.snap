// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <nav
      className="navbar navbar-inverse"
      style={
        Object {
          "marginBottom": 0,
        }
      }
    >
      <div
        className="container"
      >
        <div
          className="navbar-header"
        >
          <li
            className="navbar-brand"
            role="presentation"
            style={undefined}
          >
            <a
              action="push"
              href="#"
              onClick={[Function]}
              role="button"
            >
              Carpe Noctem
            </a>
          </li>
          <button
            className="navbar-toggle collapsed"
            onClick={[Function]}
            type="button"
          >
            <span
              className="sr-only"
            >
              Toggle navigation
            </span>
            <span
              className="icon-bar"
            />
            <span
              className="icon-bar"
            />
            <span
              className="icon-bar"
            />
          </button>
        </div>
        <div
          aria-expanded={null}
          className="navbar-collapse collapse"
        >
          <ul
            className="nav navbar-nav"
            role={null}
          >
            <li
              className=""
              role="presentation"
              style={undefined}
            >
              <a
                action="push"
                href="#"
                onClick={[Function]}
                role="button"
              >
                Home
              </a>
            </li>
          </ul>
          <ul
            className="nav navbar-nav navbar-right"
            role={null}
          >
            <li
              className=""
              role="presentation"
              style={undefined}
            >
              <a
                action="push"
                href="#"
                onClick={[Function]}
                role="button"
              >
                Simple Game
              </a>
            </li>
            <li
              className=""
              role="presentation"
              style={undefined}
            >
              <a
                action="push"
                href="#"
                onClick={[Function]}
                role="button"
              >
                Moon Shooter
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </div>
  <div
    className="row"
  />
</div>
`;
